src/sizzle.js

```diff
@@ -1596,7 +1596,7 @@ Expr = Sizzle.selectors = {
 		}),
 
 		"lt": createPositionalPseudo(function( matchIndexes, length, argument ) {
-			var i = argument < 0 ? argument + length : argument;
+			var i = argument < 0 ? argument + length : Math.min(argument, length);
```

Comment:
I like this idea, but let's reduce its size impact.
```suggestion
			var i = argument < 0 ?
				argument + length :
				argument > length ?
					length :
					argument;
```

Reply:
@gibson042 I didn't expect you to be in favor, but if we make this change, there are similar protections that can be added to `:gt` and `:eq`.

Reply:
As it turns out, the other positional pseudos are already inherently bounded. I'd be against this if it were costly, but it's actually a small fix that makes a big difference (if only in rare cases).

Reply:
@timmywil  `:gt` already has protection (if `argument` is bigger than `length`, the for condition fails at the first pass). To `:eq`, we can use this but I'm not convinced:

```JavaScript
"eq": createPositionalPseudo(function( matchIndexes, length, argument ) {
    return argument < 0 ?
            [ argument + length ] :
            argument > length ?
                    [] :
                    [ argument ];
}),
```

@gibson042 Can i rebase to do 1 commit and build dist files (_sizzle.js_, _sizzle.min.js_ and _sizzle.min.map_) ?

Reply:
> Can i rebase to do 1 commit and build dist files (<em>sizzle.js</em>, <em>sizzle.min.js</em> and <em>sizzle.min.map</em>) ?

:+1:

Reply:
- [x] Build dist files
- [x] Clean commit